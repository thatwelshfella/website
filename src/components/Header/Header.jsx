import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import './Header.less';

const Header = ({ icon, menu }) => {
  const [showMenu, setShowMenu] = useState(false);

  const toggleMobileMenu = () => {
    setShowMenu(!showMenu);
  };
  return (
    <>
      <div className="header-component">
        {icon}
        <FontAwesomeIcon icon="bars" size="lg" onClick={toggleMobileMenu} />
      </div>
      <div className={showMenu ? 'mobile-menu show' : 'mobile-menu hidden'}>
        {menu}
      </div>
    </>
  );
};

Header.propTypes = {
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  menu: PropTypes.array.isRequired,
};

export default Header;
