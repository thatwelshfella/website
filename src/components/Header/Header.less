.mobile {
  display: block;

  @media only screen and (min-width: 800px) {
    display: none;
  }
}

.desktop {
  display: block;

  @media only screen and (max-width: 800px) {
    display: none;
  }
}

.header-component {
  padding: 15px;

  border-bottom: 1px lightgrey solid;
  box-shadow: 0px 1px 15px;
  background-color: #f5f5f5;

  display: flex;
  justify-content: space-between;

  position: relative;
  z-index: 100;

  svg {
    cursor: pointer;
  }
}

.mobile-menu {
  height: 100%;

  position: fixed;
  z-index: 50;

  background-color: #f5f5f5;

  overflow: hidden;

  transition: 0.5s;

  &.show {
    width: 100%;
    display: flex;
    justify-content: center;
  }

  &.hidden {
    width: 0%;
  }
}

.menu-items {
  a {
    padding: 10px;
    color: black;
    text-decoration: none;
  }

  &.container {
    display: flex;
    flex-direction: column;

    text-align: left;

    &.hidden {
      display: none;
    }
  }
}
