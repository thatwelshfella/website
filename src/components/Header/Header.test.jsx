import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('<Header />', () => {
  const menu = [
    { link: '#home', value: 'Home' },
    { link: '#aboutMe', value: 'About me' },
    { link: '#timeline', value: 'Employment Timeline' },
    { link: '#github', value: 'Github Projects' },
  ];

  const cmp = shallow(<Header icon="LD" menu={menu} />);

  describe('@renders', () => {
    it.only('should render with mobile and desktop headers ', () => {
      expect(cmp).toMatchSnapshot();
    });
  });
});
