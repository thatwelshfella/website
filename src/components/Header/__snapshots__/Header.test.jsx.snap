// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> @renders should render with mobile and desktop headers  1`] = `
<Fragment>
  <div
    className="mobile"
  >
    <MobileHeader
      icon="LD"
      menu={
        Array [
          Object {
            "link": "#home",
            "value": "Home",
          },
          Object {
            "link": "#aboutMe",
            "value": "About me",
          },
          Object {
            "link": "#timeline",
            "value": "Employment Timeline",
          },
          Object {
            "link": "#github",
            "value": "Github Projects",
          },
        ]
      }
    />
  </div>
  <div
    className="desktop"
  >
    <DesktopHeader
      icon="LD"
      menu={
        Array [
          Object {
            "link": "#home",
            "value": "Home",
          },
          Object {
            "link": "#aboutMe",
            "value": "About me",
          },
          Object {
            "link": "#timeline",
            "value": "Employment Timeline",
          },
          Object {
            "link": "#github",
            "value": "Github Projects",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
